src/Features/CSharp/Portable/ConvertToRecord/ConvertToRecordCommon.cs

```diff
@@ -399,15 +438,19 @@ private static SyntaxList<AttributeListSyntax> GetModifiedAttributeListsForPrope
             // this variable doubles as a flag to see if we need to generate doc comments at all, as
             // if it is still null, we found no meaningful doc comments anywhere
             var exteriorTrivia = GetExteriorTrivia(typeDeclaration) ??
-                properties.SelectAsArray(GetExteriorTrivia).FirstOrDefault(t => t != null);
+                propertyResults
+                .Where(result => !result.IsInherited)
+                .Select(result => GetExteriorTrivia(result.Declaration!))
+                .FirstOrDefault(trivia => trivia != null);
 
             if (exteriorTrivia == null)
             {
                 // we didn't find any substantive doc comments, just give the current non-doc comments
                 return SyntaxFactory.TriviaList(classTrivia.Concat(propertyNonDocComments).Select(trivia => trivia.AsElastic()));
             }
 
-            var propertyParamComments = CreateParamComments(properties, exteriorTrivia!.Value, lineFormattingOptions);
+            // comments for inherited parameters go first
```

Comment:
Not in order of the parameters, whatever those are?

Reply:
That comment is actually wrong, we do go in order of the parameters. I think it was from when I was handling inherited and non-inherited parameters in separate lists, but I'll just delete it.